### PR TITLE
Calling monkey.patch_all() should patch all API methods

### PR DESCRIPTION
**Pull request by prashantmital**

### CODE REVIEW

File: bindings/python/pymongoarrow/api.py [hunk 1]
```diff
@@ -32,6 +32,10 @@
 _PATCH_METHODS = [
     'aggregate_arrow_all',
     'find_arrow_all',
+    'aggregate_pandas_all',
+    'find_pandas_all',
+    'aggregate_numpy_all',
+    'find_numpy_all',
```
Comment (ShaneHarvey) [inline bindings/python/pymongoarrow/api.py:38]: Can we add tests for this?

File: bindings/python/pymongoarrow/monkey.py [hunk 2]
```diff
@@ -11,12 +11,30 @@
...
 __all__ = [
     'patch_all'
 ]
 
 
 def patch_all():
+    """Patch all PyMongoArrow methods into PyMongo.
+
+    This function should be called **before** ``pymongo`` is imported for
+    the first time. Failing to do so might lead to unexpected results.
```
Comment (ShaneHarvey) [inline bindings/python/pymongoarrow/monkey.py:25]: Is this true? What unexpected results? AFAIK we can patch methods onto classes at anytime.

Comment (prashantmital) [reply bindings/python/pymongoarrow/monkey.py]: Removed.

File: bindings/python/pymongoarrow/api.py (hunk 1 above)
Comment (prashantmital) [reply bindings/python/pymongoarrow/api.py:38]: Done.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**